src/test/java/e2e/pages/BasePage.java

```diff
@@ -0,0 +1,49 @@
+package e2e.pages;
+
+import e2e.WebDriverProvider;
+import e2e.config.ConfigurationProvider;
+import e2e.config.TestConfiguration;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+public abstract class BasePage {
+    protected WebDriver driver;
+
+    private final TestConfiguration config;
+
+    public BasePage() throws Exception {
+        this.config = ConfigurationProvider.getConfig();
```

Comment:
Do you think the ConfigurationProvider saves parsed config somewhere after first read? It would be best to avoid parsing a file every time we create a page object